code/datums/status_effects/xeno_buffs.dm

```diff
@@ -102,33 +101,43 @@
 	///Minimum health threshold before the effect is deactivated
 	var/minimum_health
 
-/datum/status_effect/xeno_psychic_link/on_creation(mob/living/new_owner, set_duration, mob/living/carbon/target_mob, link_range, redirect_mod, minimum_health)
+/datum/status_effect/xeno_psychic_link/on_apply()
+	. = ..()
+	if(HAS_TRAIT(target_mob, TRAIT_PSY_LINKED))
+		return FALSE
+
+/datum/status_effect/xeno_psychic_link/on_creation(mob/living/new_owner, set_duration, mob/living/carbon/target_mob, link_range, redirect_mod, minimum_health, scaling = FALSE)
 	owner = new_owner
 	duration = set_duration
 	src.target_mob = target_mob
 	src.link_range = link_range
 	src.redirect_mod = redirect_mod
 	src.minimum_health = minimum_health
-	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
-	RegisterSignal(target_mob, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
+	ADD_TRAIT(target_mob, TRAIT_PSY_LINKED, REF(src))
```

Comment:
why a ref and not a defined string like every other trait

Reply:
I was thinking of making it possible to apply psychic link if you don't have the buff on yourself, but are the target of one, meaning you could have a trait from your buff and from someone else's, but I removed this option now.